By Salwa Salsabila | Universitas Dian Nuswantoro

# Arithmetic and Variables

Make calculations, and define and modify variables.

## Printing

Di Python, kita dapat memerintahkan komputer untuk menampilkan pesan dengan menggunakan fungsi *print()* dan memberikan pesan yang diinginkan di dalam tanda kurung, diapit tanda kutip. Misalnya, jika kita ingin mencetak pesan "Helo, world!", kita menulis kode berikut:

In [1]:
print("Hello, world!")

Hello, world!


Dengan menggunakan perintah print, kita memberi tahu komputer untuk mencetak atau menampilkan teks yang ada di dalam tanda kurung ganda (") ke layar. Dalam hal ini, teks yang akan dicetak adalah "Hello, world!", yang merupakan pesan yang sering digunakan dalam tutorial pemrograman sebagai langkah pertama dalam mempelajari sebuah bahasa pemrograman baru.

Jadi, ketika kita menjalankan perintah print("Hello, world!"), kita akan melihat pesan "Hello, world!" muncul di layar komputer sebagai hasil dari program yang kita tulis.

## Arithmetic

Kita bisa juga menampilkan hasil dari beberapa operasi matematika (seperti penambahan, pengurangan, perkalian, atau pembagian).

In [2]:
print(1 + 2)

3


Perintah *print(1 + 2)* dalam bahasa pemrograman Python adalah cara untuk menampilkan hasil penjumlahan dari angka 1 dan 2 ke layar komputer. Dalam kasus ini, tanda kurung digunakan untuk menunjukkan bahwa kita ingin mencetak hasil operasi matematika yang terdapat di dalamnya.

Python akan menjumlahkan angka 1 dan 2, dan output yang dihasilkan 3. Jadi, hasil dari perintah ini adalah angka 3. Ini adalah contoh sederhana dari bagaimana kita dapat menggunakan Python untuk melakukan operasi matematika dasar dan menampilkan hasilnya kepada pengguna.

In [3]:
print(9 - 5)

4


Perintah *print(9 - 5)* dalam bahasa pemrograman Python adalah cara untuk menampilkan hasil pengurangan dari angka 9 dan 5 ke layar komputer. Ketika kode ini dieksekusi, Python akan mengurangkan angka 5 dari angka 9, yaitu 4, dan kemudian mencetak hasilnya ke layar. Jadi, hasil dari perintah ini adalah angka 4

In [4]:
print(((1 + 3) * (9 - 2) / 2) ** 2)

196.0


Perintah *print(((1 + 3) * (9 - 2) / 2) ** 2)* dalam bahasa pemrograman Python adalah cara untuk menghitung hasil ekspresi matematika yang kompleks dan menampilkan hasilnya ke layar komputer. Dalam ekspresi ini, terdapat beberapa operasi matematika yang dilakukan secara berurutan.

1. (1 + 3): Operasi penjumlahan dari angka 1 dan 3, menghasilkan 4.
2. (9 - 2): Operasi pengurangan dari angka 9 dan 2, menghasilkan 7.
3. (4 * 7): Operasi perkalian dari hasil penjumlahan dan pengurangan sebelumnya, menghasilkan 28.
4. (28 / 2): Operasi pembagian hasil perkalian sebelumnya dengan angka 2, menghasilkan 14.
5. ** 2: Operasi pemangkatan hasil pembagian sebelumnya ke pangkat 2, menghasilkan 196.

Hasil akhir dari ekspresi ini adalah angka 196.

## Comments

Komentar adalah bagian dari kode yang tidak dieksekusi oleh Python. Mereka digunakan untuk memberikan penjelasan atau dokumentasi tentang apa yang dilakukan oleh baris kode yang terkait.

Contohnya, di sel kode berikut, kita mengalikan 3 dengan 2. Kita tambahkan komentar (# Multiply 3 by 2) di atas kode untuk menjelaskan apa yang dilakukan oleh kode tersebut.

In [5]:
# Multiply 3 by 2
print(3 * 2)

6


Untuk menandai bahwa suatu baris adalah komentar dalam Python, gunakan tanda pagar (#) sebagai karakter pertama.

Setelah menemukan tanda pagar, Python akan menganggap baris tersebut sebagai komentar dan akan mengabaikannya sepenuhnya. Ini penting karena Python mengikuti aturan yang sangat ketat, dan hanya akan menghasilkan kesalahan jika tidak dapat memahami kode yang diberikan.

Contohnya terlihat pada sel kode di bawah ini. Python akan menghasilkan kesalahan jika tanda pagar dihilangkan karena teks dalam komentar bukanlah kode Python yang valid dan tidak dapat diinterpretasikan dengan benar.

In [7]:
Multiply 3 by 2

SyntaxError: invalid syntax (3750420471.py, line 1)

## Variables

Variabel dalam pemrograman adalah sebuah nama atau identifikasi yang diberikan untuk menyimpan suatu nilai data dalam memori komputer. Nilai ini dapat diubah atau dimanipulasi selama eksekusi program. Setiap variabel memiliki tipe data yang menentukan jenis nilai yang dapat disimpan, seperti angka, teks, atau struktur data yang lebih kompleks. Nama variabel yang dipilih biasanya mendeskripsikan data yang disimpan, memudahkan pemahaman dan pengelolaan kode.

### Creating variables

Sel kode berikutnya membuat variabel bernama test_var dan memberinya nilai yang kita dapatkan saat kita menambahkan 5 ke 4.

Kami kemudian mencetak nilai yang ditetapkan ke variabel, yaitu 9.

In [1]:
# Create a variable called test_var and give it a value of 4+5
test_var = 4 + 5

# Print the value of test_var
print(test_var)

9


Jadi, jika mengeksekusi sel di atas, variabel test_var akan diberi nilai 9 (hasil dari 4 + 5), dan hasilnya akan dicetak ke layar sebagai “9”.

Secara umum, untuk bekerja dengan variabel, harus memulai dengan memilih nama yang ingin digunakan. Nama variabel idealnya pendek dan deskriptif. Mereka juga harus memenuhi beberapa persyaratan:

1. Mereka tidak boleh memiliki spasi (misalnya, test var tidak diperbolehkan)
2. Mereka hanya dapat menyertakan huruf, angka, dan garis bawah (misalnya, test_var! tidak diperbolehkan)
3. Mereka harus dimulai dengan huruf atau garis bawah (misalnya, 1_var tidak diperbolehkan)

Kemudian, untuk membuat variabel, perlu menggunakan = untuk menetapkan nilai yang  diinginkan.

Selalu dapat melihat nilai yang diberikan ke variabel dengan menggunakan print() dan memasukkan nama variabel dalam tanda kurung.

### Manipulating variables

Dapat juga mengubah nilai yang ditetapkan ke variabel dengan mengganti nilai sebelumnya.

Di sel kode di bawah ini, akan mengubah nilai my_var dari 3 menjadi 100.

In [2]:
# Set the value of a new variable to 3
my_var = 3

# Print the value assigned to my_var
print(my_var)

# Change the value of the variable to 100
my_var = 100

# Print the new value assigned to my_var
print(my_var)
3

3
100


3

Nilai my_var dimulai dari 3, lalu diubah menjadi 100. Saat mencetak my_var, ini akan menampilkan nilai saat ini yang ditetapkan padanya

Perhatikan bahwa secara umum, setiap kali mendefinisikan variabel dalam sel kode, semua sel kode berikutnya juga memiliki akses ke variabel tersebut. Misalnya, kita menggunakan sel kode berikutnya untuk mengakses nilai my_var (dari sel kode di atas) dan test_var (dari awal tutorial ini).

In [3]:
print(my_var)
print(test_var)

100
9


Nilai my_var adalah 100, dan nilai test_var adalah 9. Saat mencetak variabel-variabel ini, nilai yang ditetapkan saat ini akan ditampilkan.

Sel kode berikutnya memberitahu Python untuk meningkatkan nilai my_var saat ini sebesar 3.

Untuk melakukan ini, kita masih perlu menggunakan my_var = seperti sebelumnya. Dan juga sama seperti sebelumnya, nilai baru yang ingin kita berikan pada variabel ada di sebelah kanan tanda =/sign

In [4]:
# Increase the value by 3
my_var = my_var + 3

# Print the value assigned to my_var
print(my_var)

103


Nilai my_var sekarang menjadi 103. Sebelumnya 100, dan menambahkan 3 ke dalamnya. Saat mencetak my_var, ini menampilkan nilai yang diperbarui.

### Using multiple variables

Biasanya kode menggunakan banyak variabel. Hal ini sangat berguna ketika harus melakukan perhitungan panjang dengan banyak masukan.

Di sel kode berikutnya, akan menghitung jumlah detik dalam empat tahun. Perhitungan ini menggunakan lima input.

In [5]:
# Create variables
num_years = 4
days_per_year = 365 
hours_per_day = 24
mins_per_hour = 60
secs_per_min = 60

# Calculate number of seconds in four years
total_secs = secs_per_min * mins_per_hour * hours_per_day * days_per_year * num_years
print(total_secs)

126144000


Jumlah detik dalam empat tahun adalah 126.144.000.

Perhatikan bahwa penggunaan variabel sangat berguna ketika nilai input dapat berubah. Misalnya, kita ingin sedikit meningkatkan perkiraan kita dengan memperbarui nilai jumlah hari dalam setahun dari 365 menjadi 365,25, untuk memperhitungkan tahun kabisat. Kemudian kita dapat mengubah nilai yang ditetapkan ke hari_per_tahun tanpa mengubah variabel lain dan mengulangi penghitungan.

In [6]:
# Update to include leap years
days_per_year = 365.25

# Calculate number of seconds in four years
total_secs = secs_per_min * mins_per_hour * hours_per_day * days_per_year * num_years
print(total_secs)

126230400.0


Jumlah detik dalam empat tahun, terhitung tahun kabisat, adalah sekitar 126.230.400 detik.

*Note*: Penambahan 0 diakhir angka disebabkan karena pada perhitungan kedua menggunakan bilangan yang bagian pecahannya (365,25), sedangkan pada perhitungan pertama hanya mengalikan bilangan yang tidak ada bagian pecahannya. 

### Debugging

Salah satu kesalahan umum saat bekerja dengan variabel adalah kesalahan ketik yang tidak disengaja. Misalnya, jika mengeja hours_per_day as hours_per_dy, Python akan error dengan pesan NameError: name 'hours_per_dy' is not defined.

In [7]:
print(hours_per_dy)

NameError: name 'hours_per_dy' is not defined

Saat melihat NameError seperti ini, merupakan indikasi bahwa harus memeriksa bagaimana mengeja variabel yang dirujuknya sebagai "tidak ditentukan". Kemudian, untuk memperbaiki kesalahan tersebut, hanya perlu memperbaiki ejaannya.

In [8]:
print(hours_per_day)

24


Nilai jam_per_hari adalah 24.